src/Commands/Add.hs

```diff
@@ -3,28 +3,33 @@ module Commands.Add
     ) where
 
 
-import           Data.List        (lines, unlines)
-import           Entry
-import           Helpers          (getFilePath, indentedOutput, loadLines)
-import           System.Directory (removeFile, renameFile)
-import           System.IO        (readFile)
+import           Data.List (lines, unlines)
+import qualified Entry     as E
+import           Helpers   (getFilePath, indentedOutput, updateFileContent)
+import           System.IO (readFile)
 
 
--- @TODO: REFACTOR
 add :: [String] -> IO String
 add []              = return "No arguments provided.\nFor help: timetrack add --help"
 add ["-h"]          = return "usage: timetrack add YYYY-MM-DD \"message\""
 add ["--help"]      = return "usage: timetrack add YYYY-MM-DD \"message\""
 add [_]             = return "Not enough arguments provided.\nFor help: timetrack add --help"
 add [date, message] = do
     path <- getFilePath
-    entries <- fmap parseLines loadLines
-    let index = nextRowNum entries
-        newEntry = Entry { index = index, date = date, message = message }
-        updatedEntries = entries ++ [newEntry]
-        tmpPath = path ++ ".tmp"
-    writeFile tmpPath $ unlines $ fmap showOutput updatedEntries
-    removeFile path
-    renameFile tmpPath path
-    let output = indentedOutput $ showEntry newEntry
-    return $ "=> Wrote to " ++ path ++ "\n" ++ output
+    entries <- E.loadEntries path
+
+    let output = fmap E.showOutput (entries ++ [E.Entry {
+        E.index = E.nextRowNum entries,
+        E.date = date,
+        E.message = message
+    }])
+
+    msg =<< updateFileContent path (unlines output)
+
+
+msg :: (FilePath, String) -> IO String
```

Comment:
Wish this could be `-> String`, but not sure how to make it work without doing it this way and staying within the `IO` world